lager/detail/nodes.hpp

```diff
@@ -178,17 +178,19 @@ class reader_node : public reader_node_base
     {
         using namespace std;
         if (flags_ == needs_notify_flag) {
-            flags_ = no_flag;
+            flags_         = no_flag;
+            bool outermost = outermost_;
+            outermost_     = false;
             observers_(last_);
-            auto garbage = false;
             for (std::size_t i = 0, size = children_.size(); i < size; ++i) {
                 if (auto child = children_[i].lock()) {
                     child->notify();
```

Comment:
This code is dangerously not exception-safe now. If `notify` throws `outermost_` will stay false forever and no later notification would succeed. Use RAII to set the flag.

Reply:
Yep, you're absolutely right. I knew something felt off when I wrote it and I want sure how to fix it at the time.

Reply:
I decided on using a `unique_ptr` with a custom deleter to do the RAII for `outermost_`.

Reply:
While the compiler may see through the various layers of indirection, a `unique_ptr` feels a bit overkill for this...

Reply:
Also the code is incorrect, it should set it back to `outermost`, not to `true`.

Reply:
it's not more expensive than using the `this` pointer. `unique_ptr` isn't going to allocate anything, I'm just passing it a pointer to my bool flag and a custom deleter. I thought this was the canonical way to do RAII wrapping?